pkg/tfvars/openstack/openstack.go

```diff
@@ -35,7 +45,72 @@ func TFVars(masterConfig *v1alpha1.OpenstackProviderSpec, cloud string, region s
 		IngressVIP:      ingressVIP,
 		TrunkSupport:    trunkSupport,
 		OctaviaSupport:  octaviaSupport,
+		SwiftPublicURL:  swiftPublicURL,
 	}
 
 	return json.MarshalIndent(cfg, "", "  ")
 }
+
+func getSwiftPublicURL(cloud string) (string, error) {
+	var swiftPublicURL string
+	serviceCatalog, err := getServiceCatalog(cloud)
+	if err != nil {
+		return "", err
+	}
+
+	for _, svc := range serviceCatalog.Entries {
+		if svc.Type == "object-store" {
+			for _, e := range svc.Endpoints {
+				if e.Interface == "public" {
+					swiftPublicURL = e.URL
```

Comment:
shouldn't you break here? why going over the rest?

Reply:
yep, you're right